File: app/src/main/java/com/yeraygarcia/recipes/util/Debug.java
package com.yeraygarcia.recipes.util;

import android.util.Log;

public final class Debug {

    private static final String PREFIX = "YGQ: ";
    private static final boolean DEBUG = true;
    public static final boolean POPULATE_DATABASE = true;

    private Debug() {
        // avoid instantiation
    }

    public static void d(Object obj, String message) {
        if (DEBUG) {
            Log.d(PREFIX + obj.getClass().getSimpleName(), message);
        }
    }

    public static void d(String tag, String message) {
        if (DEBUG) {
            Log.d(PREFIX + tag, message);
        }
    }
}

File: app/src/main/java/com/yeraygarcia/recipes/util/Debug.kt
package com.yeraygarcia.recipes.util

import com.yeraygarcia.recipes.BuildConfig
import timber.log.Timber

object Debug {

    private const val PREFIX = "YGQ: "

    fun d(obj: Any, message: String?) {
        if (BuildConfig.DEBUG) {
            val tag = PREFIX + (obj as? String ?: obj.javaClass.simpleName)
            Timber.tag(tag).d(message)
        }
    }
}
